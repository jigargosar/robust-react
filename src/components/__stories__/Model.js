import {action} from '@storybook/addon-actions'
import {storiesOf} from '@storybook/react'
import {centerPaper} from '../../centerPaper'
import {h} from '../../hyper-script'
import {createFakeModels} from '../__fixtures__/Model.fixture'
import {ModelList} from '../Model'

const story = storiesOf('Components | ModelList', module).addDecorator(
  centerPaper,
)

story.add('basic', () =>
  h(ModelList, {
    models: createFakeModels(),
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
story.add('with primaryRenderer', () =>
  h(ModelList, {
    models: createFakeModels(),
    primaryRenderer: model => `${model.name} (id:${model.id})`,
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
