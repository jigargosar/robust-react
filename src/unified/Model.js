import {ListItemText, MenuItem} from 'material-ui'
import {storiesOf} from '../facade'
import {h} from '../hyper-script'
import {centerPaper} from './story-decorators/centerPaper'
// import 'dom-testing-library/extend-expect'
// import {render} from 'react-testing-library'
// import {describe, expect, it, specs, storiesOf} from '../../../../facade'

const ModelListItem = ({model}) => {
  return h(MenuItem, [h(ListItemText, model.name)])
}

// const createFakeModels = () => {
//   const chance = Chance(11)
//   return R.times(
//     (idx: number) => ({id: idx, name: chance.country({full: true})}),
//     3,
//   )
// }

storiesOf('Unified | Model', module)
  .addDecorator(centerPaper)
  .add('ListItem', () => {
    return h(ModelListItem, {model: {id: 0, name: 'Foo bar'}})
  })
