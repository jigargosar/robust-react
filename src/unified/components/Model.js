import Chance from 'chance'
import {List, ListItemText, MenuItem} from 'material-ui'
import {head, map, propOr, times} from 'ramda'
import {storiesOf} from '../../facade'
import {h} from '../../hyper-script'
import {centerPaper} from './story-decorators/centerPaper'
// import 'dom-testing-library/extend-expect'
// import {render} from 'react-testing-library'
// import {describe, expect, it, specs, storiesOf} from '../../../../facade'

const story = storiesOf('Unified | Model', module).addDecorator(centerPaper)

const createFakes = () => {
  const chance = Chance(11)
  return times(idx => ({id: idx, name: chance.country({full: true})}), 3)
}

const ModelListItem = ({
  model,
  primary = propOr('<primary or name not found>', 'name'),
  onClick = () => () => {},
}) => h(MenuItem, {onClick: onClick(model)}, [h(ListItemText, primary(model))])

story.add('ListItem', () => h(ModelListItem, {model: head(createFakes())}))

const ModelList = ({models, ...other}) =>
  h(List, map(model => h(ModelListItem, {model, ...other}), models))

story.add('List', () => h(ModelList, {models: createFakes()}))

export {ModelList}
