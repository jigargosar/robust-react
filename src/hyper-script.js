import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
