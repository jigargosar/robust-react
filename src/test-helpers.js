import {render} from 'react-testing-library'

export const getRendered = component => render(component).container.firstChild
