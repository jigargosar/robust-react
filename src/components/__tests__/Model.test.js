import Chance from 'chance'
import 'dom-testing-library/extend-expect'
import {times} from 'ramda'
import {h} from '../../hyper-script'
import {getRendered} from '../../test-helpers'
import {ModelList} from '../Model'

const createFakes = () => {
  const chance = new Chance(11)
  return times(idx => ({id: idx, name: chance.country({full: true})}), 3)
}

describe('Components', () => {
  describe('ModelList', () => {
    it('it should render without crashing', () => {
      expect(
        getRendered(h(ModelList, {models: createFakes()})),
      ).toMatchSnapshot()
    })
  })
})
