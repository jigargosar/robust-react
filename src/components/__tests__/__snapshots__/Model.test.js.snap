// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ModelList it should render without crashing 1`] = `
<ul
  class="MuiList-root-1 MuiList-padding-2"
>
  <li
    class="MuiButtonBase-root-17 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15 MuiMenuItem-root-5"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root-20"
    >
      <h3
        class="MuiTypography-root-26 MuiTypography-subheading-33 MuiListItemText-primary-23"
      >
        Ceuta & Melilla
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root-50"
    />
  </li>
  <li
    class="MuiButtonBase-root-17 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15 MuiMenuItem-root-5"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root-20"
    >
      <h3
        class="MuiTypography-root-26 MuiTypography-subheading-33 MuiListItemText-primary-23"
      >
        Bahamas
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root-50"
    />
  </li>
  <li
    class="MuiButtonBase-root-17 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-14 MuiListItem-button-15 MuiMenuItem-root-5"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root-20"
    >
      <h3
        class="MuiTypography-root-26 MuiTypography-subheading-33 MuiListItemText-primary-23"
      >
        American Samoa
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root-50"
    />
  </li>
</ul>
`;
